### PR TITLE
fix: do not use '.exp.sed' as redact source

### PR DESCRIPTION
**Pull request by xiangce**
- see: INSGHTCORE-227
- fix: #3858
- fix: #3885

### All Pull Requests:

Check all that apply:

* [x] Have you followed the guidelines in our Contributing document, including the instructions about commit messages?
* [x] Is this PR to correct an issue?
* [ ] Is this PR an enhancement?

### CODE REVIEW

File: insights/tests/client/data_collector/test__process_content_redaction.py
Comment (xiangce) [reply insights/tests/client/data_collector/test__process_content_redaction.py:61]: @ryan-blakley, yes, we want to do that, since the `password` redaction caused some issues (#3858, #3885, and another not-reported one in the `insalled_rpms`) and resulted in false positives of some rules. 
and, the current logic of the `.exp.sed` is not sufficient, e.g. the keyword `password` is not the only one for indicating passwords. Considering these, and we have a strict review process for Specs, We'd like to remove it from the collection. What do you think?

Comment (xiangce) [reply insights/tests/client/data_collector/test__process_content_redaction.py:61]: Ps, the `_process_content_redaction` method will be removed also in the next `Move obfuscation/reaction to core collection` (#3679). The relevant tests will also be removed then.

File: insights/client/data_collector.py
Comment (ahitacat) [inline insights/client/data_collector.py:39]: It is no based in that file anymore.

**Review by ahitacat — CHANGES_REQUESTED** [review]
There is still left the `default_sed_file` defined in the constants file.

Except for this it looks great. I'll approve it once the constant is removed.

File: insights/client/data_collector.py
Comment (xiangce) [reply insights/client/data_collector.py:39]: So I made an incorrect understanding, I thought you (the client team) might want to keep these `settings` until remove this file from the RPM package.

I'll remove them in the next commit soon.

**Comment by xiangce** [comment]
> There is still left the `default_sed_file` defined in the constants file.
> 
> Except for this it looks great. I'll approve it once the constant is removed.

So I made an incorrect understanding, I thought you (the client team) might want to keep these `settings` in the constants until remove this file from the RPM package.

I'll remove them in the next commit soon.

File: insights/client/data_collector.py
Comment (xiangce) [reply insights/client/data_collector.py:39]: Sure, I'll update this docstring. Thanks for reviewing.

**Comment by ahitacat** [comment]
> > There is still left the `default_sed_file` defined in the constants file.
> > Except for this it looks great. I'll approve it once the constant is removed.
> 
> So I made an incorrect understanding, I thought you (the client team) might want to keep these `settings` in the constants until remove this file from the RPM package.
> 
> I'll remove them in the next commit soon.

Oh, I think is no needed. 
With this PR it is not longer used in the code so I feel it shouldn't be defined in constants.

File: insights/tests/client/data_collector/test__process_content_redaction.py
Comment (ryan-blakley) [reply insights/tests/client/data_collector/test__process_content_redaction.py:61]: Gotcha yeah I noticed the issues that were caused by it. Is the plan to not merge this until the new obfuscation/redaction is merged? The reason I ask, I just don't want passwords to get stored in warehouse after making this change, and then have to purge them.

Comment (xiangce) [reply insights/tests/client/data_collector/test__process_content_redaction.py:61]: Sorry @ryan-blakley, this `.exp.sed` or the equivalent process will not be used/implemented in the new obfuscation/redaction. Our current target is to completely discard it from the core (and count on spec reviews to avoid collecting PII information) due to 1. it causes issues on specs 2. it's not sufficient. 

However, if you have come across any real cases where this "password" script works as expected, for example, actually obfuscated the password, can you share some of these cases with me? 

We have talked about the above in the sync meeting too. I believe Bob will also ask you the same for me. It won't hurt that we start another e-mail or Slack thread to continue discussing this matter.

Comment (ryan-blakley) [reply insights/tests/client/data_collector/test__process_content_redaction.py:61]: @xiangce Gotcha, I apologize I thought the current `.exp.sed` script was actively redacting passwords. Looking through several older archives I have downloaded I see how bad it is now. I did see it redacted the proxy password from the /etc/rhsm/rhsm.conf file correctly, but that's not as big of a deal. My fear was we would start collecting service user passwords, or database user passwords for like mariadb, redis, etc. But if that's not the case, then the change looks good to me!

Comment (xiangce) [reply insights/tests/client/data_collector/test__process_content_redaction.py:61]: @ryan-blakley - Thanks for the info. So I think it's required for us to go through the existing (`non-filterable`) specs to check if there are any risks of collecting the sensitive information. We can merge this PR after this check.

And I opened #3916 first for `rhsm.conf`

Comment (xiangce) [reply insights/tests/client/data_collector/test__process_content_redaction.py:61]: @ryan-blakley - FYI, I opened INSGHTCORE-236 to track the password check of existing specs.

Comment (ryan-blakley) [reply insights/tests/client/data_collector/test__process_content_redaction.py:61]: Sounds perfect to me!

**Pull request closed without merging**
